Wrong arguments to `setGridSpacing` (and others) not properly diagnosed

I received this error when building my application:
```
/home/eschnetter/src/CarpetX/Cactus/configs/sim/lib/libthorn_CarpetX.a(io_openpmd.cxx.o): In function `CarpetX::carpetx_openpmd_t::OutputOpenPMD(_cGH const*, std::vector<bool, std::allocator<bool> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
io_openpmd.cxx:(.text+0xeec): undefined reference to `openPMD::Mesh& openPMD::Mesh::setGridSpacing<unsigned long>(std::vector<unsigned long, std::allocator<unsigned long> > const&)'
```

The reason is that I accidentally passed a `vector<unsigned long>` with the grid spacing. The C++ header files accept this, but apparently the respective template instantiation is missing.

Could this be diagnosed better? For example, the C++ headers files could could check that the template argument of `setGridSpacing` is one of the supported arguments. [`std::is_floating_point`](https://en.cppreference.com/w/cpp/types/is_floating_point) would be handy for this.

The same problem exists for a few other functions as well.

--- a/include/openPMD/Mesh.hpp
+++ b/include/openPMD/Mesh.hpp
@@ -27,6 +27,7 @@
 #include <array>
 #include <ostream>
 #include <string>
+#include <type_traits>
 #include <vector>
 
 
@@ -142,7 +143,8 @@ public:
      * @param   gridSpacing     vector containing N (T) elements, where N is the number of dimensions in the simulation.
      * @return  Reference to modified mesh.
      */
-    template< typename T >
+    template< typename T,
+              typename = std::enable_if_t<std::is_floating_point< T >::value> >
     Mesh& setGridSpacing(std::vector< T > const & gridSpacing);
 
     /**
@@ -188,7 +190,8 @@ public:
      * @param   timeOffset  Offset between the time at which this record is defined and the Iteration::time attribute of the Series::basePath level.
      * @return  Reference to modified mesh.
      */
-    template< typename T >
+    template< typename T,
+              typename = std::enable_if_t<std::is_floating_point< T >::value> >
     Mesh& setTimeOffset(T timeOffset);
 
 private:

--- a/src/Mesh.cpp
+++ b/src/Mesh.cpp
@@ -142,7 +142,7 @@ Mesh::setAxisLabels(std::vector< std::string > const & als)
     return *this;
 }
 
-template< typename T >
+template< typename T, typename >
 Mesh&
 Mesh::setGridSpacing(std::vector< T > const & gs)
 {
@@ -201,7 +201,7 @@ Mesh::setUnitDimension(std::map< UnitDimension, double > const& udim)
     return *this;
 }
 
-template< typename T >
+template< typename T, typename >
 Mesh&
 Mesh::setTimeOffset(T to)
 {

